### PR TITLE
Make GitHub import errors more descriptive

### PR DESCRIPTION
**Pull request by lbogdan**
...and maybe other errors, as well?

This PR changes this GitHub import error message:

image

into this:

image

### CODE REVIEW

File: packages/app/src/app/overmind/namespaces/editor/actions.ts [hunk 1]
```diff
@@ -100,7 +100,11 @@ export const sandboxChanged: AsyncAction<{ id: string }> = withLoadApp<{
     actions.workspace.openDefaultItem();
   } catch (error) {
     state.editor.notFound = true;
-    state.editor.error = error.message;
+    let detail = error.response?.data?.errors?.detail;
```
Comment (lbogdan) [inline packages/app/src/app/overmind/namespaces/editor/actions.ts]: Again, while it works, I'm not sure it's the best approach to handle this here, but it was the easiest to get the job done. 🙂

**Comment by lbogdan** [comment]
Build for latest commit https://github.com/codesandbox/codesandbox-client/pull/3430/commits/f1cb04f33698316e266be4faa8001c5d6e61820d is at https://pr3430.build.csb.dev/s/new.

File: packages/app/src/app/overmind/namespaces/editor/actions.ts (hunk 1 above)
Comment (CompuIves) [reply packages/app/src/app/overmind/namespaces/editor/actions.ts]: That's interesting, I was thinking that our error handling logic should solve this and put the error already on `error.message`. Didn't we change something here @christianalfoni?

File: packages/app/src/app/pages/Sandbox/index.tsx [hunk 2]
```diff
@@ -77,7 +77,7 @@ export const Sandbox: React.FC<Props> = ({ match }) => {
             Something went wrong
           </div>
           <Title style={{ fontSize: '1.25rem', marginBottom: 0 }}>
-            {state.editor.error}
+            Error importing GitHub repository: {state.editor.error}
```
Comment (CompuIves) [inline packages/app/src/app/pages/Sandbox/index.tsx]: We should probably change this message based on the `isGithub` variable name.

File: packages/app/src/app/overmind/namespaces/editor/actions.ts (hunk 1 above)
Comment (lbogdan) [reply packages/app/src/app/overmind/namespaces/editor/actions.ts]: As far as I remember, we always had that 422 error message.

Comment (CompuIves) [reply packages/app/src/app/overmind/namespaces/editor/actions.ts]: Hmyes, but there was a time that we actually showed the error. There has been a regression long ago that made it look like this cryptic error.

File: packages/app/src/app/pages/Sandbox/index.tsx (hunk 2 above)
Comment (lbogdan) [reply packages/app/src/app/pages/Sandbox/index.tsx]: Good catch, updated!

**Pull request merged**